x-pack/plugin/sql/qa/src/main/resources/datetime.csv-spec

```diff
@@ -277,6 +277,177 @@ null    | 2004-04-30 00:00:00.000Z
+10036    | 1959-08-10 00:00:00.000Z | 1992-01-03 00:00:00.000Z | 33         | -130         | 389        | -1691     | 11834    | -17040960 | 1022457600
+10037    | 1963-07-22 00:00:00.000Z | 1990-12-05 00:00:00.000Z | 27         | -109         | 329        | -1429     | 9998     | -14397120 | 863827200
+10038    | 1960-07-20 00:00:00.000Z | 1989-09-20 00:00:00.000Z | 29         | -116         | 350        | -1523     | 10654    | -15341760 | 920505600
+10039    | 1959-10-01 00:00:00.000Z | 1988-01-19 00:00:00.000Z | 29         | -113         | 339        | -1477     | 10337    | -14885280 | 893116800
+10040    | null                     | 1993-02-14 00:00:00.000Z | null       | null         | null       | null      | null     | null      | null
+10041    | null                     | 1989-11-12 00:00:00.000Z | null       | null         | null       | null      | null     | null      | null
+10042    | null                     | 1993-03-21 00:00:00.000Z | null       | null         | null       | null      | null     | null      | null
+;
+
+selectDiffWithComplexExpressions1
+schema::gender:s|birth_date:ts|hire_date:ts|diff:i
+SELECT gender, birth_date, hire_date, TIMESTAMPDIFF('months', birth_date + CASE WHEN gender = 'M' THEN INTERVAL 10 years
+WHEN gender = 'F' THEN INTERVAL -10 years ELSE INTERVAL 12 years END, hire_date + INTERVAL 10 month) AS diff
+FROM test_emp WHERE (diff BETWEEN 150 AND 250) OR diff > 500 ORDER BY emp_no LIMIT 10;
+
+  gender  |      birth_date          |       hire_date          | diff
+----------+--------------------------+--------------------------+-----
+M         | 1959-12-03 00:00:00.000Z | 1986-08-28 00:00:00.000Z | 210
+F         | 1953-04-20 00:00:00.000Z | 1989-06-02 00:00:00.000Z | 564
+F         | 1957-05-23 00:00:00.000Z | 1989-02-10 00:00:00.000Z | 511
+F         | 1952-04-19 00:00:00.000Z | 1985-02-18 00:00:00.000Z | 524
+null      | 1963-06-01 00:00:00.000Z | 1989-08-24 00:00:00.000Z | 180
+null      | 1956-02-12 00:00:00.000Z | 1987-03-11 00:00:00.000Z | 239
+null      | 1959-08-19 00:00:00.000Z | 1987-07-02 00:00:00.000Z | 201
+M         | 1960-02-20 00:00:00.000Z | 1988-02-10 00:00:00.000Z | 226
+F         | 1953-09-29 00:00:00.000Z | 1989-12-17 00:00:00.000Z | 565
+F         | 1958-09-05 00:00:00.000Z | 1997-05-19 00:00:00.000Z | 594
+;
+
+selectDiffWithComplexExpressions2
+schema::languages:byte|first_name:s|gender:s|birth_date:ts|hire_date:ts|diff:s
+SELECT languages, first_name, gender, birth_date, hire_date,
+CAST(DATE_DIFF(CASE WHEN gender = 'M' THEN CONCAT(gender, 'onths') WHEN gender = 'F' THEN NULL ELSE 'quarter' END,
+    hire_date + INTERVAL 20 month, birth_date - interval 50 month) AS string) AS diff
+FROM test_emp WHERE languages >= 3 AND first_name LIKE '%y%' ORDER BY diff ASC, languages DESC;
+
+   languages  |   first_name  |  gender  |      birth_date          |       hire_date          |  diff
```

Comment:
Hm, something seems off here indeed, will check it.

Reply:
It's a cast to string that's why we get this ordering, but to avoid future confusion changed the cast to long: https://github.com/elastic/elasticsearch/pull/47920/commits/97171038d7e60c57e874758bf0a824da07288e84

Reply:
I've already checked that when I came into that but my memory becomes weaker... :(